dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java

```diff
@@ -391,6 +254,174 @@ private void updateAuditStatus() throws DotPublisherException, DotDataException
+            // If bundle was installed in all groups
+            PushPublishLogger.log(this.getClass(), "Status Update: Success");
+            bundleStatus = Status.SUCCESS;
+            pubAuditAPI.updatePublishAuditStatus(bundleAudit.getBundleId(), bundleStatus, localHistory);
+            pubAPI.deleteElementsFromPublishQueueTable(bundleAudit.getBundleId());
+        } else if ( groupPushStats.countGroupPublishing() == endpointTrackingMap.size() ) {
+            // If bundle is still publishing in all groups
+            bundleStatus = Status.PUBLISHING_BUNDLE;
+            pubAuditAPI.updatePublishAuditStatus(bundleAudit.getBundleId(), bundleStatus, localHistory);
+        } else {
+            // Otherwise, just keep trying to publish the bundle
+            bundleStatus = Status.WAITING_FOR_PUBLISHING;
+            pubAuditAPI.updatePublishAuditStatus(bundleAudit.getBundleId(), bundleStatus, localHistory);
+        }
+
+		Logger.info(this, "===========================================================");
+		Logger.info(this, String.format("For bundle '%s':", bundleAudit.getBundleId()));
+		if (!bundleStatus.equals(PublishAuditStatus.Status.PUBLISHING_BUNDLE)
+				&& !bundleStatus.equals(PublishAuditStatus.Status.WAITING_FOR_PUBLISHING)) {
+			Logger.info(this, String.format("-> Re-try attempts: %d", localHistory.getNumTries()));
+		}
+		Logger.info(this, String.format("-> Status:          %s", bundleStatus.toString()));
+		Logger.info(this, "===========================================================");
+	}
+
+	private GroupPushStats getGroupStats(Map<String, Map<String, EndpointDetail>> endpointTrackingMap) {
+		GroupPushStats groupPushStats = new GroupPushStats();
+
+		for (Map<String, EndpointDetail> group : endpointTrackingMap.values()) {
+            boolean isGroupOk = false;
+            boolean isGroupPublishing = false;
+            boolean isGroupFailed = false;
+            for (EndpointDetail detail : group.values() ) {
+                if ( detail.getStatus() == Status.SUCCESS.getCode() ) {
+                    isGroupOk = true;
+                } else if ( detail.getStatus() == Status.PUBLISHING_BUNDLE
+                        .getCode() ) {
+                    isGroupPublishing = true;
+                } else if ( detail.getStatus() == Status.FAILED_TO_PUBLISH
+                        .getCode() ) {
+                    isGroupFailed = true;
+                }
+            }
+            if ( isGroupOk ) {
+                groupPushStats.increaseCountGroupOk();
+            }
+            if ( isGroupPublishing ) {
+                groupPushStats.increaseCountGroupPublishing();
+            }
+            if ( isGroupFailed ) {
+                groupPushStats.increaseCountGroupFailed();
+            }
+        }
+
+        return groupPushStats;
+	}
+
+	private void updateLocalEndpointDetailFromRemote(PublishAuditHistory localHistory, String groupID, String endpointID, PublishAuditHistory remoteHistory) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 155 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)